lib/membrane/rtp/packet.ex

```diff
@@ -100,17 +129,39 @@ defmodule Membrane.RTP.Packet do
   def parse(_binary, _parse_payload?), do: {:error, :malformed_packet}
 
   defp parse_header_extension(binary, header_present?)
-  defp parse_header_extension(binary, false), do: {:ok, {nil, binary}}
+  defp parse_header_extension(binary, false), do: {:ok, {[], binary}}
 
   defp parse_header_extension(
-         <<profile_specific::binary-size(2), data_len::16, data::binary-size(data_len)-unit(32),
+         <<_profile_specific::binary-size(2), data_len::16, data::binary-size(data_len)-unit(32),
            rest::binary>>,
          true
        ) do
-    extension = %Header.Extension{profile_specific: profile_specific, data: data}
-
-    {:ok, {extension, rest}}
+    extensions = parse_extension_data(data, [])
+    extensions = Enum.reverse(extensions)
+    {:ok, {extensions, rest}}
   end
 
   defp parse_header_extension(_binary, true), do: :error
+
+  defp parse_extension_data(<<>>, acc), do: acc
+
+  defp parse_extension_data(data, acc) do
+    case parse_one_byte_header(data) do
+      {data, rest} -> parse_extension_data(rest, [data | acc])
+      rest -> parse_extension_data(rest, acc)
+    end
+  end
+
+  defp parse_one_byte_header(<<extension_header::bits-size(8), extensions::binary>>) do
```

Comment:
Could you do that `if` statement as two different function pattern matches as you are already matching on it?